creator-node/src/serviceRegistry.js

```diff
@@ -44,64 +52,174 @@ class ServiceRegistry {
     await this.blacklistManager.init()
 
     if (!config.get('isUserMetadataNode')) {
-      this.libs = await initAudiusLibs()
+      this.libs = await this._initAudiusLibs()
+
+      this.URSMRegistrationManager = new URSMRegistrationManager(this.nodeConfig, this.libs)
+
+      /**
+       * Do not await any of the below processes as they will not complete until after server is up.
+       * Server is started after serviceRegistry.initServices() is called.
+       */
 
-      this.snapbackSM = new SnapbackSM(this.libs)
-      await this.snapbackSM.init()
+      this._recoverNodeL1Identity()
```

Comment:
instead of having each one of these spin until the previous is finished, why not have a single task that calls them in a row and just fire that off?

Reply:
not saying this is better/worse, just noticing that each of them has its own `while` clause that spins until the prior condition is completed

Reply:
way cleaner, fixed